guillotina/db/transaction_manager.py

```diff
@@ -138,7 +138,9 @@ def lock(self):
                         raise
             except Exception:
                 # failsafe terminate to make sure connection is cleaned
-                if txn._db_conn is not None:
+                if txn._db_conn is None:
+                    raise
+                if txn._db_conn._con is None:
```

Comment:
What is this case? Connection holder doesn't actually have a connection?

How does the asyncpg pool handle this case?

Should we still call `await self._storage.terminate(txn._db_conn)` on it?

Reply:
I think that this conditional is impossible to happend:
```
if txn._db_conn is None:
          raise
```
but i'm not sure about this:
```
if txn._db_conn._con is None:
    raise
```
Basically, it depends on which is the exception that raised `asyncpg.pool.release()`. I guess the connection might be already terminated or not...

Reply:
alright, as long as it's safe. Was just wonder if it was possible to be closed by not released or something weird like that.